projects/js-packages/svelte-data-sync-client/src/initializeClient.ts

```diff
@@ -0,0 +1,106 @@
+import { z } from 'zod';
+import { API } from './API';
+import { API_Endpoint } from './Endpoint';
+import { SyncedStore } from './SyncedStore';
+import { ValidatedValue } from './types';
+
+export function getValidatedValue< T extends z.ZodSchema >(
+	namespace: string,
+	valueName: string,
+	valueSchema: T
+): ValidatedValue< T > {
+	const validator = z.object( {
+		value: valueSchema,
+		nonce: z.string(),
+	} );
+
+	/**
+	 * This function handles fetching nonces if they're there
+	 * But it shouldn't trigger errors if the nonce is missing.
+	 * Allowing the client application decide how to handle
+	 * missing nonces.
+	 *
+	 * This is useful, for example, if the `valueSchema` is set to something
+	 * like `z.string().optional()`. In this case, the application might be
+	 * okay with a missing value, so a missing nonce would trigger an unexpected
+	 * error.
+	 */
+	const source = window[ namespace ][ valueName ]
+		? window[ namespace ][ valueName ]
+		: { value: undefined, nonce: '' };
+
+	const result = validator.parse( source );
+
+	return result as ValidatedValue< T >;
+}
+
+// `window.[name].rest_api` is a special value that's required for the API to work.
+// It's populated by the `wp-js-data-sync` package and contains the REST API endpoint and nonce.
+function setupRestApi( namespace: string ) {
+	const api = new API();
+	try {
+		const { value, nonce } = getValidatedValue( namespace, 'rest_api', z.string().url() );
+		api.initialize( value, nonce );
+	} catch ( e ) {
+		// eslint-disable-next-line no-console
+		console.error(
+			`Failed to connect to REST API because of an invalid "window.${ namespace }.rest_api" value:\n`,
+			`	Expected Example: `,
+			{ value: 'https://example.com/wp-json', nonce: 'abc123' },
+			`\n	Received Value: `,
+			window[ namespace ]?.rest_api,
+			'\n\nError originated from: \n ',
+			e
+		);
+	}
+
+	return api;
+}
+
+/**
+ * Initialize the client-side data sync.
+ *
+ * Usage:
+ *
+ *  1. Which namespace to use?
+ * 	2. Create a Store that's going to sync.
+ *  3. Reference $widget in Svelte component to use it as a regular Svelte Store.
+ * 	4. This will update the Svelte Store and POST `false` to `example.com/wp-json/jetpack-boost/widget-status`
```

Comment:
This numbering mixes spaces and tabs, causing a janky in-and-out line-up on any tab width other than 4.

"Which namespace to use?" -- should this explain that the namespace is the name of the global/window variable that these values go in? Maybe give an example.

Why do we want to POST false to widget-status?

I like that this is followed by a js example. Maybe move these numbers into the js sample as ... comments within the comment? Is that silly?

Reply:
There's something janky going on GitHub I think, looked fine in the editor. I've updated it and clarified it anyway.

As for the JS example, it's subtle - but each of the steps is numbered ( `1: , 2: ` ) and references a comment above. I prefer to keep them separate so that if you just want a code example, you can get it without comments getting in the way.